src/layout/layout.rs

```diff
@@ -109,7 +110,7 @@ thread_local! {
 #[derive(Debug, Default, Clone, Eq, PartialEq, Hash)]
 pub struct Layout {
     direction: Direction,
-    constraints: Vec<Constraint>,
+    constraints: SmallVec<Constraint, 5>,
```

Comment:
fwiw, we won't need this change here because `Layout` is cached.

Reply:
Hmmm, I don't think that's true. Every time you call vertical or horizontal there's an allocation: https://github.com/ratatui-org/ratatui/blob/57dc90736f2daa624ceb6649e90c9cb794cc235d/src/layout/layout.rs#L163

Reply:
My bad!

Reply:
No worries! :) Also turns out just accessing the cache clones the layout (which I don't think can be fixed sadly): https://github.com/ratatui-org/ratatui/blob/57dc90736f2daa624ceb6649e90c9cb794cc235d/src/layout/layout.rs#L582

And there's some locking that doesn't make sense since it's a thread-local which I've fixed here: https://github.com/ratatui-org/ratatui/pull/1245

Reply:
yeah, I just noticed the clone there too.

Thanks for all the PRs!